### ISSUE TITLE
@SuppressAnnotation to support list of violations

### ISSUE THREAD

**Issue by nitirajrathore** [policeman-tools/forbidden-apis#157]
Hi, 

We are using suppression [like this](https://github.com/elastic/elasticsearch/blob/4c981ff4bfc250080d521af105b5e8589c9fc517/core/src/main/java/org/elasticsearch/common/SuppressForbidden.java) in elasticsearch project. But we are faced with a complication. We are extended a forbidden class and to make it work we used @SuppressAnnotation at class level. But because of this class level annotation all other violations of the class are also skipped, like use of System.out or String.format(...) with default locale. 

What we need is to suppress only a particular violation. Something like

```
public @interface SuppressForbidden {
    String reason();
    String[] voilations;
}
```

So that only the listed violations are suppressed and others still get caught by forbidden-apis.

Is such feature already available? or can it be implemented in future release?

**Comment by mikemccand** [comment]
+1, I've hit this limitation before too (working on Lucene's sources!).  It'd be great if `forbiddenapis` had more granularity ... maybe it does and we just don't know how @uschindler?

**Comment by uschindler** [comment]
Hi,
The problem is that we just have a list of signatures, but no categories. The forbidden signatures may come from different sources, but we don't know their names. It's just an input stream coming from Maven/Gradle or classloader.

For bundled signatures we have some name, but before adding suppression (like javac suppressions) all signatures need to be categorized.

After that the suggestion here might be implemented, but there must be a way to tell forbiddenapis which attribute of the Annotation triggers the exclusion. Because we decided to use any Annotation defined in the project. This prevents to require forbiddenapis at compile time in classpath.

In short not sure if this could be implemented. For next major version we may change signatures file format to add categories (in same way like the error messages per signature).

The example given above is one problem that cannot be solved by refactoring - so sorry! But in general the best work around us to use method level annotations and refactor code to have the suppressions only on arbitrary added method bodies around the problematic code.

If the

**Comment by bmhm** [comment]
Hi, I'd love to see this feature, too! Can anyone of you create a PR? :)
